src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_CollectionExpression.cs

```diff
@@ -831,12 +831,19 @@ private BoundExpression CreateAndPopulateArray(BoundCollectionExpression node, A
 
             RewriteCollectionExpressionElementsIntoTemporaries(elements, numberIncludingLastSpread, localsBuilder, sideEffects);
 
-            // int index = 0;
-            BoundLocal indexTemp = _factory.StoreToTemp(
-                _factory.Literal(0),
-                out assignmentToTemp);
-            localsBuilder.Add(indexTemp);
-            sideEffects.Add(assignmentToTemp);
+            // When there are no spread elements, we can use constant indices for better codegen.
+            // When there are spread elements, we need a mutable index variable for the CopyTo optimization.
+            BoundLocal? indexTemp = null;
+
+            if (numberIncludingLastSpread > 0)
```

Comment:
Ah. Looking at the code from above, I recall now, that this optimization won't ever kick in, until we extend it so that normal elements before the first spread element use constant indices. Preconditions earlier on this code path, make it so the condition on this line will always be true, as currently written.

Sorry for the runaround.